### ISSUE TITLE
Questions about the matching loss

### ISSUE THREAD

**Issue by gabrielhuang** [dddzg/up-detr#10]
Hi there, I have a few important questions about multiple-query-patch pretraining, which I could not find answer to in the paper.

1) Is there a reason why during multiple-query-patch pretraining, all predictions are matched with all query patches instead of matching each prediction only within its own group? 
2) Is the binary ground-truth (c_i) just always 1, or does it take different values, depending on which group was matched to it? In other words, will predictions from a given group be penalized for predicting the patch from a different group, if they turn out to be matched with that other group?
3) Also what is the meaning/interpretation of $\widehat{c}_i=1$ during multiple-query-patch pretraining? 

Thanks for the amazing work by the way.

**Comment by dddzg** [comment]
Thanks for your attention!
1. You are right. I also have ever considered this problem beforing experiments. However, after experiments, we find that the model has learn to predict from the same group. So, we did not change the matching implementation. 
2. Yes, there is a mis-matching theoretically. But, I did not observe a noticable problem with current setting.
3. On the one way, with label c=1, we can get the model prediction result in the inference stage(like the visualization).  One the other way, with c=1 and c=0, we can construct a many-one matching mechanism and the model can learn to a NMS-like mechanism during the pre-training.
